metrics/metric_observer.go

```diff
@@ -6,32 +6,15 @@ import (
 
 	"github.com/openshift-online/async-routine"
 	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
 )
 
 var _ async.RoutinesObserver = (*metricObserver)(nil)
 
-type metricObserver struct{}
-
-var (
-	runningRoutines = prometheus.NewGauge(
-		prometheus.GaugeOpts{
-			Name: "async_routine_manager_routines",
-			Help: "Number of running routines.",
-		},
-	)
-
-	runningRoutinesByName = prometheus.NewGaugeVec(
-		prometheus.GaugeOpts{
-			Name: "async_routine_manager_routine_instances",
-			Help: "Number of running instances of a given routine.",
-		},
-		[]string{"routine_name", "data"},
-	)
-)
-
-func init() {
-	prometheus.MustRegister(runningRoutines)
-	prometheus.MustRegister(runningRoutinesByName)
+type metricObserver struct {
+	r                     prometheus.Registerer
```

Comment:
Please rename to have a more meaningful field name. `r` may have different meanings in some contexts and may get abmbigues.